src/commands/cmd_server.cc

```diff
@@ -1221,6 +1221,39 @@ class CommandAnalyze : public Commander {
   std::vector<std::string> command_args_;
 };
 
+class CommandReset : public Commander {
+ public:
+  Status Execute(Server *srv, Connection *conn, std::string *output) override {
+    // 1. Discards the current MULTI transaction block, if one exists.
+    if (conn->IsFlagEnabled(Connection::kMultiExec)) {
+      conn->ResetMultiExec();
+    }
+    // 2. Unwatches all keys WATCHed by the connection.
+    srv->ResetWatchedKeys(conn);
+    // 3. Disables CLIENT TRACKING, if in use. (not yet supported)
+    // 4. Sets the connection to READWRITE mode.
+    // 5. Cancels the connection's ASKING mode, if previously set. (not yet supported)
+    // 6. Sets CLIENT REPLY to ON. (not yet supported)
+    // 9. Exits MONITOR mode, when applicable.
+    if (conn->IsFlagEnabled(Connection::kMonitor)) {
+      conn->Owner()->QuitMonitorConn(conn);
+    }
+    // 10. Aborts Pub/Sub's subscription state (SUBSCRIBE and PSUBSCRIBE), when appropriate.
+    if (conn->SubscriptionsCount() != 0) {
+      conn->UnsubscribeAll();
+    }
+    if (conn->PSubscriptionsCount() != 0) {
+      conn->PUnsubscribeAll();
+    }
```

Comment:
The connection will not receive any commands after entering the monitor or subscribe mode, so I guess it should be unable to cancel them here.

Reply:
Oh I didn't know that. I thought only redis-cli would behave this way. Do you have any idea how do bypass this?

Reply:
@chrisxu333 Maybe I was wrong. We didn't remove the READABLE event from fd after entering the monitor/subscriber mode on the server side. So it may be good for the current implementation. You can print some logs or use debug mode to confirm this.

Reply:
Here is the log and corresponding code:
![image](https://github.com/apache/kvrocks/assets/44099579/a8a0561a-6b54-4d05-a912-dfa76c18e604)
![image](https://github.com/apache/kvrocks/assets/44099579/217a5203-48da-49fc-a36a-171dd725f232)

Reply:
Sure, thank you!